app/pods/search/newsletter-infos/route.js

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable class-methods-use-this */
+import Route from '@ember/routing/route';
+import { isEmpty, isPresent } from '@ember/utils';
+import moment from 'moment';
+import search from 'frontend-kaleidos/utils/mu-search';
+import Snapshot from 'frontend-kaleidos/utils/snapshot';
+import { inject as service } from '@ember/service';
+
+export default class AgendaitemSearchRoute extends Route {
+  @service metrics;
+  queryParams = {
+    page: {
+      refreshModel: true,
+      as: 'pagina',
+    },
+    size: {
+      refreshModel: true,
+      as: 'aantal',
+    },
+    sort: {
+      refreshModel: true,
+      as: 'sorteer',
+    },
+  };
+
+  textSearchFields = Object.freeze(['title', 'subTitle', 'richtext']);
+
+  constructor() {
+    super(...arguments);
+    this.lastParams = new Snapshot();
+  }
+
+  postProcessAgendaitems(newsletter) {
+    const agendaitems = newsletter.attributes.agendaitems;
+    if (agendaitems) {
+      if (Array.isArray(agendaitems)) {
+        for (const agendaitem of agendaitems) {
+          if (agendaitem['nextVersionId'] == null) {
+            // there is no next version = latest agendaitem
+            return (newsletter.attributes.latestAgendaitem = agendaitem);
+          }
+        }
+      }
+      return (newsletter.attributes.latestAgendaitem = agendaitems);
+    }
+  }
+
+  postProcessDecisions(newsletter) {
+    const decisions = newsletter.attributes.decisions;
+    if (decisions) {
+      if (Array.isArray(decisions)) {
+        // TODO for now, if there are multiple decisions, we just grab the first one
+        return (newsletter.attributes.decision = decisions.firstObject);
+      }
+      return (newsletter.attributes.decision = decisions);
+    }
+  }
+
+  postProcessMandatees(newsletter) {
+    const mandatees = newsletter.attributes.latestAgendaitem.mandatees;
+    if (mandatees) {
+      if (Array.isArray(mandatees)) {
+        const sortedMandatees = mandatees.sortBy('priority');
+        const mandateeMap = [];
+        // For each mandatee, we have to combine the first and family name
```

Comment:
L65-73 is render-logica (die naar de template mag), niet?

Reply:
Ik had het hier gestoken omdat ik dan uiteindelijk gewoon een string heb dat ik moet tonen in de frontend.
Ongeacht of ik nu 1 minister object heb, of een array van minister objecten.
Dan blijft de frontend logica ook simpeler.

Reply:
Da's volgens mij niet de goede redenering. Dat kan je doortrekken naar het samenstellen van een html-string om de template logica simpeler te houden. Ik zou hier enkel zorgen dat de `mandatees` steeds een array is. Wanneer je het gevoel krijgt dat de template complex/moeilijk leesbaar wordt, moet je opdelen in componenten.

Reply:
Logica zit nu in de template